bindings/python/src/decoders.rs

```diff
@@ -591,9 +592,71 @@ pub fn decoders(m: &Bound<'_, PyModule>) -> PyResult<()> {
     m.add_class::<PyBPEDecoder>()?;
     m.add_class::<PyCTCDecoder>()?;
     m.add_class::<PySequenceDecoder>()?;
+    m.add_class::<PyDecodeStream>()?;
     Ok(())
 }
 
+/// Class needed for streaming decode
+///
+#[pyclass(module = "tokenizers.decoders", name = "DecodeStream")]
+#[derive(Clone)]
+pub struct PyDecodeStream {
```

Comment:
intersting, so we don't use the rust `DecodeStream` object. I am guessing it's for ownership reasons? Otherwise we needs to wrap the `PyDecodeStream {stream: DecodeStream}` with arc?

Reply:
Arc doesn't save us. The issue is with the borrow of the tokenizer's lifetime.

You can technically reborrow in every call in rust too, it's just not very "rusty".
For Python, we need to get access to the tokenizer on every call, and cloning into an Arc feels super wasteful (and breaks every update you might do on the tokenizer afterwards).

This seems innocuous enough since currently users have to hold already a reference to the tokenizer anyway.